Src/AutoFixture/EmailAddressLocalPart.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Represents the local part of the email address, defined as everything up to, but not including the @ sign.  
+    /// The requirements enforced on the local part are defined in RFC 3696.  The regular expression for validating 
+    /// the local part borrowed from Phil Haack at 
+    /// http://haacked.com/archive/2007/08/21/i-knew-how-to-validate-an-email-address-until-i.aspx/
+    /// </summary>
+    public class EmailAddressLocalPart : IEquatable<EmailAddressLocalPart>
+    {
+        private static string pattern = @"^(?!\.)(""([^""\r\\]|\\[""\r\\])*""|"
+                                               + @"([-A-Za-z0-9!#$%&'*+/=?^_`{|}~]|(?<!\.)\.)*)(?<!\.)$";
+        private readonly string localPart;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EmailAddressLocalPart"/> class. Throws ArgumentNullException
+        /// if localPart is null or empty.  Throws ArgumentException if localPart is longer than 64 characters or
+        /// does not conform to the local pat requirements per RFC 3696. 
+        /// </summary>
+        /// <param name="localPart">The local part.</param>        
+        public EmailAddressLocalPart(string localPart)
+        {
+            if (string.IsNullOrEmpty(localPart))
+            {
+                throw new ArgumentNullException(localPart);
```

Comment:
Considering you covered this in your blog http://blog.ploeh.dk/2014/11/18/the-isnullorwhitespace-trap/ definitely not, a failure on my part.  I will modify it accordingly to only throw `ArgumentNullException` if the argument is truly null, otherwise `ArgumentException`.